exec/vector/src/main/java/org/apache/drill/exec/vector/BitVector.java

```diff
@@ -285,35 +285,55 @@ public void transferTo(BitVector target) {
 
   public void splitAndTransferTo(int startIndex, int length, BitVector target) {
     assert startIndex + length <= valueCount;
-    int firstByte = getByteIndex(startIndex);
-    int byteSize = getSizeFromCount(length);
-    int offset = startIndex % 8;
-    if (offset == 0) {
+    int firstByteIndex = getByteIndex(startIndex);//byte offset of the first src byte
+    int bytesToRead = getSizeFromCount(length); //src bytes to read (including start/end bytes that might not be fully copied)
```

Comment:
bytesToRead is also being used for setting target byte so is it better to use numOfBytes or something like that?

Reply:
Agreed. Changing it to 'numBytesHoldingSourceBits'